src/org/zaproxy/zap/extension/websocket/utility/WebSocketUtils.java

```diff
@@ -40,5 +42,31 @@ public static String encodeWebSocketKey(String key) {
             return null;
         }
     }
-
-}
+    
+    /**
+     * Generate a Sec-WebSocket-key
+     *
+     * @return new Sec-WebSocket-key
+     */
+    public static String generateSecWebSocketKey() {
+        byte[] random = new byte[16];
+        Random rand = new Random();
+        rand.nextBytes(random);
+        return Base64.getEncoder().encodeToString(random);
+    }
+    
+    /**
+     * Given a web socket key, Generate our accept key
+     */
+    public static String getAcceptKey(String strWebSocketKey) throws NoSuchAlgorithmException {
```

Comment:
We are already doing this in `encodeWebSocketKey`, worth adding the JavaDoc to that one.

Reply:
I didn't notice that, sorry

Reply:
No worries, I didn't either when in the other PR. The commit message (and PR title) should be tweaked though.